cmd/verifydependencies/verifydependencies.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// verify that dependencies are up-to-date across different files
+package main
+
+import (
+	"flag"
+	"log"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+)
+
+type dependencies struct {
+	Dependencies []*dependency `yaml:"dependencies"`
+}
+
+type dependency struct {
+	Name     string   `yaml:"name"`
+	Version  string   `yaml:"version"`
+	RefPaths []string `yaml:"refPaths"`
+}
+
+func main() {
+
+	flag.Parse()
+
+	args := flag.Args()
+
+	if len(args) == 0 {
+		log.Fatalf("usage: verifydependency <file>")
+	}
+	externalDepsFilePath := args[0]
+	externalDepsFile, err := ioutil.ReadFile(externalDepsFilePath)
+	if err != nil {
+		panic(err)
+	}
+
+	externalDeps := &dependencies{}
+
+	err = yaml.Unmarshal(externalDepsFile, externalDeps)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, dep := range externalDeps.Dependencies {
+		for _, path := range dep.RefPaths {
+			fileBytes, err := ioutil.ReadFile(path)
+			if err != nil {
+				log.Fatalf("error opening file %v : %v", path, err)
+			}
+			if !strings.Contains(string(fileBytes), dep.Version) {
```

Comment:
there a no stable ways of doing this without having custom logic per file, but maybe we can have at least so indicators per file for what constant / string to look at.

with the basic "Contains" check, a file could have multiple versions versions and maybe match something else that is not the desired version to match.